src/controllers/controllermanager.cpp

```diff
@@ -353,6 +369,11 @@ void ControllerManager::openController(Controller* pController) {
         m_pConfig->setValue(ConfigKey(
             "[Controller]", presetFilenameFromName(pController->getName())), 1);
     }
+
+    // TODO(Tomasito) Remove duplicate code (see ControllerManager::closeController())
+    if (dynamic_cast<void*>(pController) == dynamic_cast<void*>(getKeyboardController())) {
```

Comment:
It should works without any cast. If not use a static_cast<Controller*> for the right one.